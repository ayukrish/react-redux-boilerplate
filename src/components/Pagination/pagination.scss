@import '../../styles/variable.scss';

.pagination {
  width: 100%;
  justify-content: flex-end;
  margin: 20px 0;
  button {
    cursor: pointer;
    padding: 16px 20px;
    background: transparent;
    margin: 0;
    border: 1px solid $yellowtext;
    color: $yellowtext;
    margin-right: 20px;
    &:disabled {
      cursor: not-allowed;
      color: rgb(170, 170, 170);
      border: 1px solid rgb(170, 170, 170);
    }
    &:last-child {
      margin-right: 0;
    }
  }
}
