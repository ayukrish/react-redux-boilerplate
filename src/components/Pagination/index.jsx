import React from 'react';
import PropTypes from 'prop-types';
import * as style from './pagination.scss';

const propTypes = {
  contentLength: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired,
  limit: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired
};

const Pagination = ({
  contentLength, currentPage, limit, onChange
}) => {
  const showPagination = contentLength <= limit;
  return (
    <div className={`${style.pagination} flex`}>
      {showPagination && (
        <>
          <button
            type="button"
            id="prev"
            disabled={currentPage === 1}
            onClick={event => {
              event.stopPropagation();
              onChange(currentPage - 1);
            }}
          >
            Prev
          </button>
          <button
            type="button"
            id="next"
            disabled={contentLength < limit}
            onClick={event => {
              event.stopPropagation();
              onChange(currentPage + 1);
            }}
          >
            Next
          </button>
        </>
      )}
    </div>
  );
};

Pagination.propTypes = propTypes;
// Pagination.defaultProps = defaultProps;
export default Pagination;
