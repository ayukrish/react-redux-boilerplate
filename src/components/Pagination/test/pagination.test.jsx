import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

// Components
import Pagination from '../index';

describe('Pagination Test Suite', () => {
  const onChange = jest.fn(() => {});

  it('check for showPagination', () => {
    const output = shallow(
      <Pagination
        contentLength={11}
        currentPage={1}
        limit={10}
        onChange={onChange}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const output = shallow(
      <Pagination
        contentLength={9}
        currentPage={1}
        limit={10}
        onChange={onChange}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('simulate click on prev button', () => {
    const output = shallow(
      <Pagination
        contentLength={9}
        currentPage={1}
        limit={10}
        onChange={onChange}
      />
    );
    output.find('#prev').simulate('click', {
      stopPropagation: () => {}
    });
    expect(onChange).toHaveBeenCalled();
  });

  it('simulate click on next button', () => {
    const output = shallow(
      <Pagination
        contentLength={9}
        currentPage={1}
        limit={10}
        onChange={onChange}
      />
    );
    output.find('#next').simulate('click', {
      stopPropagation: () => {}
    });
    expect(onChange).toHaveBeenCalled();
  });
});
