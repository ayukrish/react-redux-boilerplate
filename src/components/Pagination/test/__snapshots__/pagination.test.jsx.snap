// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Test Suite check for showPagination 1`] = `
<div
  className="undefined flex"
/>
`;

exports[`Pagination Test Suite renders correctly 1`] = `
<div
  className="undefined flex"
>
  <button
    disabled={true}
    id="prev"
    onClick={[Function]}
    type="button"
  >
    Prev
  </button>
  <button
    disabled={true}
    id="next"
    onClick={[Function]}
    type="button"
  >
    Next
  </button>
</div>
`;
